es/contract/compiler.js

```diff
@@ -77,13 +93,48 @@ async function validateByteCodeAPI (bytecode, source, options = {}) {
 
 async function compileContractAPI (code, options = {}) {
   this.isInit()
-  return this.http.post('/compile', { code, options: this.prepareCompilerOption(options) }, options)
-    .then(({ bytecode }) => bytecode)
+  try {
+    console.log('Compiling')
+    console.log(options)
+    const eopts = this.erlpsPrepareCompilerOption(options)
+    console.log(showTerm(eopts))
+    const a = PS['Aeso.Compiler'].erlps__from_string__2([jsStrToErlStr(code), eopts]).value0[1]
+    console.log(showTerm(a))
+    const b = PS['Aeser.Contract.Code'].erlps__serialize__1([a])
+    // console.log(showTerm(b))
+    const c = PS['Aeser.Api.Encoder'].erlps__encode__2([mkAtom('contract_bytearray'), b])
+    // console.log(showTerm(c))
+    const d = erlStrToJSStr(erlBinToErlStr(c))
+    console.log('ErlPS: ', d)
+    this.http.post('/compile', { code, options: this.prepareCompilerOption(options) }, options)
+      .then(({ bytecode }) => { console.log('HTTP: ', bytecode); return bytecode })
+    return d
+  } catch (e) {
+    console.log(e)
+  }
 }
 
 async function contractGetACI (code, options = {}) {
   this.isInit()
-  return this.http.post('/aci', { code, options: this.prepareCompilerOption(options) }, options)
+  try {
+    /*
+      #{encoded_aci => lists:last(JsonACI),
+      external_encoded_aci => lists:droplast(JsonACI),
+      interface   => StringACI}
+    */
+    const eopts = this.erlpsPrepareCompilerOption(options)
+    const jsonAci = PS['Aeso.Aci'].erlps__contract_interface__3([mkAtom('json'), jsStrToErlStr(code), eopts]).value0[1]
+    const stubAci = PS['Aeso.Aci'].erlps__render_aci_json__1([jsonAci]).value0[1]
+    const ret = { interface: erlStrToJSStr(erlBinToErlStr(stubAci)) }
+    // TODO: implement erlang__atom_to_binary__2
+    // console.log(JSON.parse(erlTermToJSONBIN(json_aci).value0.toString()))
+    console.log(ret)
+    const a = await this.http.post('/aci', { code, options: this.prepareCompilerOption(options) }, options)
```

Comment:
heheh, one literally calls an emulated http endpoint ?

Reply:
No it's not emulated - this is only to compare the results of the JS compiler with the HTTP one